### PR TITLE
Fix for Property Dialog's validation of the Project ID

### PR DESCRIPTION
**Pull request by nbashirbello**
Fix for #761

The App Engine Standard Deployment property (preference) page allows the project id to be empty while the deployment wizard doesn't allow deployment without the project id specified (not empty).

### CODE REVIEW

**Comment by nbashirbello** [comment]
@elharo

**Comment by elharo** [comment]
Please merge with master

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java [hunk 1]
```diff
@@ -93,11 +93,14 @@
   private DataBindingContext bindingContext;
 
   private Runnable layoutChangedHandler;
+  private boolean childOfPropertyDialog = false;
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java]: The name childOfPropertyDialog seems rather specific. Maybe allowEmptyProjectId?

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java]: In that case, we need `allowEmptyAccount` as well, which doesn't sound like a good idea.

File: plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdInputValidator.java [hunk 2]
```diff
@@ -20,7 +25,8 @@ public IStatus validate(Object input) {
 
   private IStatus validateString(String value) {
     if (value.isEmpty()) {
-      return ValidationStatus.error(Messages.getString("project.id.empty")); //$NON-NLS-1$
+      return emptyStringAllowed ?  ValidationStatus.ok() :
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectIdInputValidator.java]: Extra space after `?`.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java (hunk 1 above)
Comment (nbashirbello) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java]: Why would we need the allowEmptyAccount?

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java]: Not in this PR, but we need it very soon when we fix #759. (Note that #742 added an account selector to the deploy preference.) We want the account choice to be an optional setting, for the same reason as project ID.

**Comment by elharo** [comment]
suggest allowEmptyProjectId --> !requireProjectId

**Pull request merged**